### PR TITLE
python3Packages.mlxtend: fix tests

### PR DESCRIPTION
**Pull request by elohmeier**
###### Description of changes

Updated the pytest configuration & added missing dependencies to fix the mlxtend build.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/mlxtend/default.nix [hunk 1]
```diff
@@ -24,28 +24,48 @@ buildPythonPackage rec {
...
+
+    # image tests download files over the network
+    "--ignore=mlxtend/image"
+
+    # tests that import deprecated sklearn datasets
+    "--ignore=mlxtend/evaluate/f_test.py"
+    "--ignore=mlxtend/feature_selection/tests/test_column_selector.py"
+    "--ignore=mlxtend/feature_selection/tests/test_exhaustive_feature_selector.py"
+    "--ignore=mlxtend/feature_selection/tests/test_sequential_feature_selector.py"
+    "--ignore=mlxtend/feature_selection/tests/test_sequential_feature_selector_feature_groups.py"
```
Comment (fabaff) [inline pkgs/development/python-modules/mlxtend/default.nix:38]: Using `disabledTestPaths` is the modern way ([17.27.1.2.5. Using pytestCheckHook](https://nixos.org/manual/nixpkgs/stable/#user-guide)).

```suggestion
  pytestFlagsArray = [
    "-sv"
  ];
  
  disabledTestPaths = [
    # image tests download files over the network
    "mlxtend/image"
    # tests that import deprecated sklearn datasets
    "mlxtend/evaluate/f_test.py"
    "mlxtend/feature_selection/tests/test_column_selector.py"
    "mlxtend/feature_selection/tests/test_exhaustive_feature_selector.py"
    "mlxtend/feature_selection/tests/test_sequential_feature_selector.py"
    "mlxtend/feature_selection/tests/test_sequential_feature_selector_feature_groups.py"
  ];
```

Comment (elohmeier) [reply pkgs/development/python-modules/mlxtend/default.nix]: Thanks, updated!

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 217278` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python310Packages.mlxtend</li>
  </ul>
</details>

**Comment by elohmeier** [comment]
Thanks, fixed.

**Comment by GetPsyched** [comment]
Hey, uh, any updates on this PR?

**Comment by elohmeier** [comment]
I have no use for mlxtend currenty, so I can't really test/proceed. Feel free to pick it up.

**Comment by GetPsyched** [comment]
Tried to pick it up but `tensorflow` errored out.

**Pull request closed without merging**